misc/python/materialize/cli/scratch/__main__.py

```diff
@@ -9,25 +9,51 @@
 
 import argparse
 
+import shtab
+
 from materialize.cli.scratch import create, destroy, login, mine, push, sftp, ssh
 
 
 def main() -> None:
     parser = argparse.ArgumentParser("scratch")
-    subparsers = parser.add_subparsers(dest="subcommand", required=True)
-    for name, configure, run in [
-        ("login", login.configure_parser, login.run),
-        ("create", create.configure_parser, create.run),
-        ("mine", mine.configure_parser, mine.run),
-        ("ssh", ssh.configure_parser, ssh.run),
-        ("sftp", sftp.configure_parser, sftp.run),
-        ("destroy", destroy.configure_parser, destroy.run),
-        ("push", push.configure_parser, push.run),
+    subparsers = parser.add_subparsers(
+        dest="subcommand", required=True, description="", help="subparsers"
+    )
+    for name, configure, run, description in [
+        ("login", login.configure_parser, login.run, "Log in to AWS SSO"),
+        (
+            "create",
+            create.configure_parser,
+            create.run,
+            "Create a new scratch instance",
+        ),
+        ("mine", mine.configure_parser, mine.run, "Show active scratch instance"),
+        ("ssh", ssh.configure_parser, ssh.run, "Connect to scratch instance via ssh"),
+        (
+            "sftp",
+            sftp.configure_parser,
+            sftp.run,
+            "Connect to scratch instance via sftp",
+        ),
+        (
+            "destroy",
+            destroy.configure_parser,
+            destroy.run,
+            "Destroy a scratch instance",
+        ),
+        (
+            "push",
+            push.configure_parser,
+            push.run,
+            "Push current HEAD (or a specific git commit) to scratch instance",
+        ),
     ]:
-        s = subparsers.add_parser(name)
+        s = subparsers.add_parser(name, description=description, help=description)
         configure(s)
         s.set_defaults(run=run)
 
+    shtab.add_argument_to(parser, ["-c", "--generate-completion"], parent=parser)  # type: ignore[no-untyped-call]
```

Comment:
Super minor, but can we run this in subparser mode for consistency with the other commands?

```diff
diff --git a/misc/python/materialize/cli/scratch/__main__.py b/misc/python/materialize/cli/scratch/__main__.py
index c11e0f79f..b5e670f1c 100644
--- a/misc/python/materialize/cli/scratch/__main__.py
+++ b/misc/python/materialize/cli/scratch/__main__.py
@@ -47,13 +47,17 @@ def main() -> None:
             push.run,
             "Push current HEAD (or a specific git commit) to scratch instance",
         ),
+        (
+            "gen-completion",
+            lambda p: shtab.add_argument_to(p, "generate-completion", parent=parser),
+            lambda: None,
+            "Generate shell completion script",
+        ),
     ]:
         s = subparsers.add_parser(name, description=description, help=description)
         configure(s)
         s.set_defaults(run=run)
 
-    shtab.add_argument_to(parser, ["-c", "--generate-completion"], parent=parser)  # type: ignore[no-untyped-call]
-
     args = parser.parse_args()
     args.run(args)
 
```

Reply:
Oh, also, I try very hard to stay away from `# type: ignore` comments! Here's a stub file you can add instead:

```diff
diff --git a/misc/python/stubs/shtab.pyi b/misc/python/stubs/shtab.pyi
new file mode 100644
index 000000000..7970f48d0
--- /dev/null
+++ b/misc/python/stubs/shtab.pyi
@@ -0,0 +1,10 @@
+from argparse import ArgumentParser
+from typing import List, Optional, Union
+
+def add_argument_to(
+    parser: ArgumentParser,
+    option_string: Union[str, List[str]] = ...,
+    help: str = ...,
+    parent: Optional[ArgumentParser] = ...,
+    preamble: str = ...,
+) -> None: ...

```

Reply:
Oooooh that's sweet, thanks for the tip. everything about the python >2.7 ecosystem is new and shiny to me 😁